File: programers_Lv0/분수의 덧셈.js
function solution(numer1, denom1, numer2, denom2) {
  var answer = [];

  let numer = numer1 * denom2 + numer2 * denom1; // 분자
  let denom = denom1 * denom2; // 분모

  // 최대 공약수 구하기
  const gcd = (a, b) => (a % b === 0 ? b : gcd(b, a % b));

  const g = gcd(denom, numer);

  answer.push(numer / g);
  answer.push(denom / g);
  return answer;
}

console.log(solution(1, 2, 3, 4));
console.log(solution(9, 2, 1, 3));

File: programers_Lv0/분수의 덧셈.py
from fractions import Fraction

def solution(numer1, denom1, numer2, denom2):
    answer = []
    numer = numer1 * denom2 + numer2 * denom1
    denom = denom1 * denom2

    f = Fraction(numer, denom)

    answer = [f.numerator, f.denominator]

    return answer

print(solution(1, 2, 3, 4))
print(solution(9, 2, 1, 3))